support/db/main.go

```
@@ -163,8 +169,46 @@ func pingDB(db *sqlx.DB) error {
 	return errors.Wrapf(err, "failed to connect to DB after %v attempts", maxDBPingAttempts)
 }
 
+type ClientConfig struct {
+	Key   string
+	Value string
+}
+
+func StatementTimeout(timeout time.Duration) ClientConfig {
+	return ClientConfig{
+		Key:   "statement_timeout",
+		Value: strconv.FormatInt(timeout.Milliseconds(), 10),
+	}
+}
+
+func IdleTransactionTimeout(timeout time.Duration) ClientConfig {
+	return ClientConfig{
+		Key:   "idle_in_transaction_session_timeout",
+		Value: strconv.FormatInt(timeout.Milliseconds(), 10),
+	}
+}
+
+func augmentDSN(dsn string, clientConfigs []ClientConfig) string {
+	parsed, err := url.Parse(dsn)
+	if err != nil || parsed.Scheme == "" {
```

Is parsing expected to fail in some cases? when? shouldn't we just fail if that's the case?

(I think a commend is in order)

---

good point, returning error here would maintain most closely the current/expected behavior, rather than new outcome of attempting fix, less code to introduce here too?

---

I added some comments. Basically, I check that we're given a url connection string (e.g. `postgres://localhost`) and if that's not the case then I assume it must be a space separated connection string. It's possible that the connection string is completely invalid but I don't attempt to validate it here and instead I will let it fail when sql.Open() is called